plugins/net_plugin/net_plugin.cpp

```diff
@@ -2412,8 +2416,29 @@ namespace eosio {
                fc_elog( logger, "Error accepting connection: ${m}", ("m", ec.message()));
                // For the listed error codes below, recall start_listen_loop()
                switch (ec.value()) {
+                  case EMFILE: // same as boost::system::errc::too_many_files_open
+                  {
+                     // no file descriptors available to accept the connection. Wait on async_timer
+                     // and retry listening
+                     if (!accept_error_timer) {
+                        try {
+                           accept_error_timer = std::make_unique<boost::asio::deadline_timer>(my_impl->thread_pool->get_executor());
```

Comment:
It seems like it'd be easier to create the timer at the same time the thread pool is created, so you don't need this try/catch/quit here.

fwiw I think this is something unfortunate about the interface of `named_thread_pool`. I would have liked the ability to use it like:
```c++
struct stuff {
   named_thrad_pool pool{"stuff"};
   boost::asio::deadline_timer timer{pool.get_executor()};
};
```
This way no optionals are needed for the thread pool nor consumers of its io_context.

Bad sadly this isn't doable with the current design; I think we'd need to completely disallow auto start & move the number of threads to `start()`

Reply:
This makes that possible: https://github.com/AntelopeIO/leap/pull/632

Reply:
That still requires the number of threads at ctor time, which means it typically can't be used as a member in a plugin_impl because the number of threads typically isn't known until plugin_startup time after configs have been parsed due to it being configurable. It seems like it'd need to be `start(num_threads)` or such instead.

Reply:
Ah, good point, I'll make that change.

Reply:
Thanks @spoonincode and @heifner , I made the change to create the timer at the same time the thread pool is created, and it is indeed simpler.